app/src/components/common/GuestApplicationButtons.tsx

```diff
@@ -0,0 +1,69 @@
+import {Button, Stack} from '@mui/material';
+import {useFormikContext} from 'formik';
+import {useNavigate} from 'react-router-dom';
+
+export const NavButtons = ({
```

Comment:
Something I would consider doing is simplifying the props passed into this component. Instead of passing in `step, setStep, and stepToRouteMapping` we could move these handlers to the parent component and pass them in instead. That way we can pass in props like `next, prev, saveAndExit`. I think passing in these handlers instead of the pieces required for composing them is a little clearer to the developer and makes the components easier to test and refactor.

Reply:
Made the changes Requested. Let me know what you think!

Reply:
Looks good! Thanks @JpadillaCoding